halo2_proofs/src/plonk/assigned.rs

```diff
@@ -440,15 +528,31 @@ mod proptests {
         }
     }
 
+    prop_compose! {
+        fn arb_operators(num_unary: usize, num_binary: usize)(
+            unary in vec(select(UNARY_OPERATORS), num_unary),
+            binary in vec(select(BINARY_OPERATORS), num_binary),
+        ) -> Vec<Operator> {
+            unary.into_iter()
+                .map(Operator::Unary)
+                .chain(binary.into_iter().map(Operator::Binary))
+                .collect()
+        }
+    }
+
     prop_compose! {
         fn arb_testcase()(
-            num_operations in 1usize..5,
+            num_unary in 0usize..5,
+            num_binary in 0usize..5,
         )(
             values in vec(
                 prop_oneof![Just(Assigned::Zero), arb_trivial(), arb_rational()],
-                num_operations + 1),
-            operations in vec(select(OPERATIONS), num_operations),
-        ) -> (Vec<Assigned<Fp>>, Vec<Operation>) {
+                // Ensure that:
+                // - we have at least one value to apply unary operators to.
+                // - we can apply every binary operator pairwise sequentially.
+                cmp::max(if num_unary > 0 { 1 } else { 0 }, num_binary + 1)),
```

Comment:
In the case where `num_unary == num_binary == 0`, do we need any values?

Reply:
It makes the rest of the rest easier to write if we always have at least one value (so we always have something to check at the end).